pkg/infrastructure/clusterapi/clusterapi.go

```diff
@@ -53,7 +53,7 @@ func InitializeProvider(platform Provider) infrastructure.Provider {
 // Provision creates cluster resources by applying CAPI manifests to a locally running control plane.
 //
 //nolint:gocyclo
-func (i *InfraProvider) Provision(dir string, parents asset.Parents) ([]*asset.File, error) {
+func (i *InfraProvider) Provision(ctx context.Context, Sdir string, parents asset.Parents) ([]*asset.File, error) {
```

Comment:
I guess it's a typo?
```suggestion
func (i *InfraProvider) Provision(ctx context.Context, dir string, parents asset.Parents) ([]*asset.File, error) {
```